### PR TITLE
Fix clippy checks for some files

### PR DESCRIPTION
**Pull request by jkitman**
I got the `cargo clippy +nightly` to pass, adding that command to `scripts/final-check.sh`

Also updated the GH actions to include `clippy --lib --bins --tests --examples --workspace -- -D warnings`

### CODE REVIEW

File: minimint/tests/fixture/mod.rs [hunk 1]
```diff
@@ -387,12 +391,12 @@ impl FederationTest {
     pub async fn run_consensus_epochs(&self, epochs: usize) {
         for _ in 0..(epochs) {
             for server in &self.servers {
-                let mut s = server.borrow_mut();
-                let height = s.consensus.wallet.consensus_height();
-                let consensus_outcome = s.outcome_receiver.recv().await.expect("other thread died");
+                let consensus_outcome = block_on(server.borrow_mut().outcome_receiver.recv())
+                    .expect("other thread died");
+                let height = server.borrow().consensus.wallet.consensus_height();
```
Comment (Maan2003) [inline minimint/tests/fixture/mod.rs:396]: please put the refcell borrow across await point changes in separate commit.

File: scripts/final-checks.sh [hunk 2]
```diff
@@ -5,7 +5,8 @@
 
 set -e
 
-cargo clippy --workspace -- -D warnings
+cargo +nightly clippy --lib --bins --tests --examples --workspace -- -D warnings
+cargo fmt
 cargo fmt --all -- --check
```
Comment (Maan2003) [inline scripts/final-checks.sh:10]: why both `cargo fmt` and `cargo fmt --all -- --check`?

Comment (jkitman) [reply scripts/final-checks.sh:10]: True, I'll remove the check, just keep `cargo fmt --all`

**Review by Maan2003 — APPROVED** [review]

File: crypto/tbs/src/lib.rs [hunk 3]
```diff
@@ -282,10 +282,9 @@ mod tests {
         let sigs = sks
             .iter()
             .enumerate()
-            .map(|(idx, sk)| (idx, sign_blinded_msg(bmsg, *sk)))
-            .collect::<Vec<_>>();
```
Comment (elsirion) [inline crypto/tbs/src/lib.rs:286]: I think this change will make it `n` times slower. Tbf, it's probably bad API design from my side.

`combine_valid_shares` takes an `IntoIter` argument where `IntoIter::Iter: Clone` and uses it as a random access iterator by cloning the iter and calling `nth`. Afaik stable rust doesn't have a `RandomAccessIterator` trait, maybe I should try to impl it (either locally or upstream).

Long story short, you are probably doing a 4ms sign operation over and over again by not collecting here.

**Review by elsirion — CHANGES_REQUESTED** [review]
Thx a lot! I think I found one regression due to an API quirk of the `tbs` crate (didn't benchmark it yet though, but am pretty sure).

File: crypto/tbs/src/lib.rs (hunk 3 above)
Comment (elsirion) [reply crypto/tbs/src/lib.rs:286]: Or I could just require a slice, but that felt lame xD

Comment (jkitman) [reply crypto/tbs/src/lib.rs:286]: I think I understand...please see the latest commit to see if it solves the issue.

File: crypto/tbs/src/lib.rs [hunk 4]
```diff
@@ -168,15 +168,19 @@ where
     I: IntoIterator<Item = (usize, BlindedSignatureShare)>,
     I::IntoIter: Clone + ExactSizeIterator,
```
Comment (Maan2003) [inline crypto/tbs/src/lib.rs:169]: let also remove `Clone + ExactSizeIterator` then

**Review by elsirion — APPROVED** [review]

**Pull request merged**